### DISCUSSION TITLE
TextMate style combo box completion in snippets

### DISCUSSION THREAD

**Discussion by vigo** [zed-industries/zed#19950] [Feature Requests]
```
{
    "demo snippet": {
        "prefix": "demo",
        "body": "${1:Text}",
        "description": "Demo..."
    }
}
```

when  i hit `demo<TAB>` it works but is there any way to add multiple options as TextMate do:

`${1|Text,foo,bar,baz|}`

so, generated dynamic options as select box?

<img width="437" alt="image" src="https://github.com/user-attachments/assets/97833da0-bb93-40e6-ac59-47fbee6c2658">

**Comment by eliduke** [comment]
I am wondering similar things along these lines! If I should create my own discussion here, I am happy to do so.

I would love to build out a more robust TextMate extension that comes with all of the amazing tab triggers and autocompletes.

For example, in TextMate I could type `div`, hit tab, and then it would populate with this text:

<img width="160" alt="Screenshot 2024-12-07 at 9 46 22 AM" src="https://github.com/user-attachments/assets/19c12d65-125a-496f-bfb1-cdc327cd628c">

If I hit tab again, it then does this:

<img width="158" alt="Screenshot 2024-12-07 at 9 46 33 AM" src="https://github.com/user-attachments/assets/33fc6d06-b938-48b1-a7bc-c369d8f6d8f1">

It is very handy because if don't want the `id` at all, I just hit delete, if I want to change the value of the `id` I hit tab again and then can type in whatever value I want.

Continuing along those lines, if I press `ctrl-shift-<` it populates with this text:

<img width="89" alt="Screenshot 2024-12-07 at 9 51 15 AM" src="https://github.com/user-attachments/assets/7a4ff6bf-45e3-4c8d-b592-eb74917eded3">

And if I start typing, it fills in the value for both the open and close tags:

<img width="203" alt="Screenshot 2024-12-07 at 9 52 15 AM" src="https://github.com/user-attachments/assets/6ba09912-0cbb-409a-9b22-28ff4a18d272">

Does kind of functionality exist in the custom bindings / snippet configurations?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
